src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1618,6 +1624,31 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RemoveInvalidSr
     return error;
 }
 
+/*
+ * @brief This is a utility function to remove all Thread client srp host and services
+ * established between the device and the srp server (in most cases the OTBR).
+ * The calling task is blocked until OnSrpClientNotification which confims the client received the request.
+ * The blocking mechanism is defined by the platform implementation of `WaitOnSrpClearAllComplete` and `NotifySrpClearAllComplete`
+ *
+ * Note: This function is meant to be used during the factory reset sequence.
+ *
+ */
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_ClearAllSrpHostAndServices()
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    if (!mIsSrpClearAllRequested)
```

Comment:
nit: check this condition after locking the stack?

Reply:
Yes I can do that. I thought reading might not be too risky but might as well do it in the lock